--- a/sshd.c
+++ b/sshd.c
@@ -122,6 +122,10 @@
 #include "ssh-sandbox.h"
 #include "version.h"
 #include "ssherr.h"
+#include <tcpd.h>
+#include <syslog.h>
+int allow_severity;
+int deny_severity;
 
 /* Re-exec fds */
 #define REEXEC_DEVCRYPTO_RESERVED_FD	(STDERR_FILENO + 1)
@@ -1985,7 +1989,22 @@ main(int ac, char **av)
 #ifdef SSH_AUDIT_EVENTS
 	audit_connection_from(remote_ip, remote_port);
 #endif
-
+allow_severity = options.log_facility|LOG_INFO;
+deny_severity = options.log_facility|LOG_WARNING;
+/* Check whether logins are denied from this host. */
+if (packet_connection_is_on_socket()) {
+        struct request_info req;
+        
+        request_init(&req, RQ_DAEMON, __progname, RQ_FILE, sock_in, 0);
+        fromhost(&req);
+        
+        if (!hosts_access(&req)) {
+                debug("Connection refused by tcp wrapper");
+                refuse(&req);
+                /* NOTREACHED */
+                fatal("libwrap refuse returns");
+        }
+}
 	/* Log the connection. */
 	laddr = get_local_ipaddr(sock_in);
 	verbose("Connection from %s port %d on %s port %d",
