--- a/netlib.c
+++ b/netlib.c
@@ -147,6 +147,7 @@ int udp_session(struct vtun_host *host)
      struct sockaddr_in saddr; 
      short port;
      int s,opt;
+     int iptos = 160;
 
      if( (s=socket(AF_INET,SOCK_DGRAM,0))== -1 ){
         vtun_syslog(LOG_ERR,"Can't create socket");
@@ -155,6 +156,7 @@ int udp_session(struct vtun_host *host)
 
      opt=1;
      setsockopt(s, SOL_SOCKET, SO_REUSEADDR, &opt, sizeof(opt)); 
+     setsockopt(s, IPPROTO_IP,IP_TOS, &iptos, sizeof(iptos));
     
      /* Set local address and port */
      local_addr(&saddr, host, 1);
--- a/tunnel.c
+++ b/tunnel.c
@@ -126,6 +126,8 @@ int tunnel(struct vtun_host *host)
 
 	   opt=1;
 	   setsockopt(host->rmt_fd,IPPROTO_TCP,TCP_NODELAY,&opt,sizeof(opt) );
+	   opt=160;
+	   setsockopt(host->rmt_fd,IPPROTO_IP,IP_TOS,&opt,sizeof(opt) );
 
 	   proto_write = tcp_write;
 	   proto_read  = tcp_read;
